src/coins.cpp

```diff
@@ -197,13 +197,9 @@ bool CCoinsViewCache::BatchWrite(CoinsViewCacheCursor& cursor, const uint256 &ha
                 // and mark it as dirty.
                 itUs = cacheCoins.try_emplace(it->first).first;
                 CCoinsCacheEntry& entry{itUs->second};
-                if (cursor.WillErase(*it)) {
-                    // Since this entry will be erased,
-                    // we can move the coin into us instead of copying it
-                    entry.coin = std::move(it->second.coin);
-                } else {
-                    entry.coin = it->second.coin;
-                }
+                cursor.WillErase(*it)
+                    ? (entry.coin = std::move(it->second.coin))
+                    : (entry.coin = it->second.coin);
```

Comment:
You're right, added the following to the description of the pr:
> To avoid repetition and make the diff trivial between the two branches (calling the copy vs move constructors), this expression was originally written as a ternary, which unfortunately introduced an additional copy operation (and was reverted to a verbose if statement with a comment).
This change attempts to restore the signal to noise ratio of such a simple expression while retaining its performance.

Reply:
I don't understand the reasoning to be honest. This seems to be just a cosmetic refactor, and I am not sure if it is really clearer, or if this was desired by other reviewers. The commit message also makes no mention of why the move constructors are now defaulted, or why that is required.

Reply:
@TheCharlatan +1

Also, the current code is clearer for me, especially due to the comment.

Reply:
The commens duplicates what the code already stated, but seems there's an agreement that this isn't an improvement, so I'll just close the PR.